src/lsp/lp_lsp.ml

```diff
@@ -55,11 +55,33 @@ let doc_table : (string, _) Hashtbl.t = Hashtbl.create 39
 let completed_table : (string, Lp_doc.t) Hashtbl.t = Hashtbl.create 39
 
 (* Notification handling; reply is optional / asynchronous *)
-let do_check_text ofmt ~doc =
+let do_check_text ?(reply=true) ofmt ~doc =
   let doc, diags = Lp_doc.check_text ~doc in
   Hashtbl.replace doc_table doc.uri doc;
   Hashtbl.replace completed_table doc.uri doc;
-  LIO.send_json ofmt @@ diags
+  if reply then
+    LIO.send_json ofmt @@ diags
+
+let do_debugFlags ofmt params =
+  let flags = string_field "flags" params in
+  LIO.log_error "debugFlags" ("flags="^flags);
+  let open Console in
+  let logger_keys = List.map (fun {logger_key; _} -> logger_key) !loggers in
+  (* let verbose = int_field "verbose" params in *)
+  Timed.Time.restore @@ Pure.get_initial_time ();
+  Console.set_debug false @@ String.of_list logger_keys;
+  Console.set_default_debug flags;
+  (* Console.set_default_verbose verbose; *)
+  Pure.set_initial_time ();
```

Comment:
I am worried about this part of the code. I believe that set_initial_time should be called only once. Resetting time erase the environment, that is, everything that was read before! Why do you need to reset time here?

Reply:
I change the default values of debug flags, set initial time (to set the default values of debug flags) then check the opened files again to generate the log messages according to the newly set flags. 
Also, I just realized that I forgot to implement the verbose part :/  will do it today

Reply:
But we may not want to get all the debug messages from the beginning. Some times we want those messages only on a specific command. I think that we need to think more about how we want to handle debug flags first.